Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/Constraints/FaceUserConstraint.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    /// <summary>
+    /// Component for fixing the rotation of a manipulated object such that
+    /// it always faces or faces away from the user
+    /// </summary>
+    public class FaceUserConstraint : TransformConstraint
+    {
+        #region Properties
+
+        [SerializeField]
+        [Tooltip("Option to use this constraint to face away from the user")]
+        private bool faceAway = false;
+
+        /// <summary>
+        /// If true, this will constrain rotation to face away from the user
+        /// </summary>
+        public bool FaceAway
+        {
+            get => faceAway;
+            set => faceAway = value;
+        }
+
+        public override TransformFlags ConstraintType => TransformFlags.Rotate;
+
+        #endregion Properties
+
+        #region Public Methods
+
+        /// <summary>
+        /// Updates an rotation so that its facing the camera
+        /// </summary>
+        public override void ApplyConstraint(ref MixedRealityTransform transform)
+        {
+            Vector3 directionToTarget = transform.Position - CameraCache.Main.transform.position;
+            transform.Rotation = Quaternion.LookRotation(faceAway ? -directionToTarget : directionToTarget);
```

Comment:
The old code that handled faceAway actually had directionToTarget (not the negative version of it) - i.e. if you see on the old line 650). I'm not an expert here so maybe there are some new changes between those two sections, just wanted to check if there was an error before (or now) or something else in between changed up.

Reply:
Agree with will -- reading the code it seems this shoudl be flipped but probably good to just verify the behavior because I could be wrong:

```suggestion
            transform.Rotation = Quaternion.LookRotation(faceAway ? directionToTarget : -directionToTarget);
```

Reply:
Oh thanks for catching this. I actually noticed that the logic in `ManipulationHandler` was incorrect but I didn't fix it properly for this constraint and forgot to fix for `ManipulationHandler`. `directionToTarget` is pointing away from the camera, not towards it. If we want a vector from the target to the camera we should have:
```csharp
Vector3 directionToTarget = CameraCache.Main.transform.position - transform.Position
```
Then we could use:
```csharp
transform.Rotation = Quaternion.LookRotation(faceAway ? directionToTarget : -directionToTarget);
```
I've confirmed that `ManipulationHandler` options do the opposite of what you'd expect. Here are the options I used for `ManipulationHandler`:
![image](https://user-images.githubusercontent.com/47415945/70794742-4b121e80-1d96-11ea-9f88-b1107978a887.png)

And here's the behaviour:
![face_user_wrong](https://user-images.githubusercontent.com/47415945/70795071-ed320680-1d96-11ea-8cf7-ab2554f7eebd.gif)

I'll fix this for both this constraint and `ManipulationHandler`

Reply:
On a second look, this isn't actually incorrect, it just so happens that most objects in the HandInteractionExample scene have their "front" on the "back" side of the object:
![image](https://user-images.githubusercontent.com/47415945/70814953-7f043880-1dc4-11ea-937a-980e5eac67c2.png)
When we do `FaceUser`/`FaceAwayFromUser` it's the blue axis that we're pointing towards/away from the user. That's why in `ManipulationHandler` the behaviour looks backwards while the logic looks more correct.

I'm not familiar with which axes generally point where on object models, so I'm not sure whether we should fix this for correct looking behaviour or for correct logic.